### DISCUSSION TITLE
How to control multi-processing / threading in pipelines

### DISCUSSION THREAD

**Discussion by treya-lin** [pyannote/pyannote-audio#1802] [Q&A] [answered]
Hi I am trying the speaker-diarization pipelines to detect speaker numbers, and I noticed that it can be very cpu hungry and each time I run it it creates a lot of processes and I can see many many processes in `htop`. My saw that there is the num_workers param that can control the processes used, but only when we call the classes like Segmentation, VAD etc...

here is how I use the model.
```
checkpoint_path = "/data/resources/models/speaker-diarization-3.1/config.yaml"
pipeline = Pipeline.from_pretrained(
            checkpoint_path
    )
pipeline.to(torch.device(device))

diarization = pipeline(audio)

        speakers = set()
        for turn, _, speaker in diarization.itertracks(yield_label=True):
            start = int(round(turn.start,3)*1000)
            end = int(round(turn.end,3)*1000)
            speaker = int(speaker.split("_")[-1]) # SPEAKER_00, SPEAKER_01
            speakers.add(speaker)

```

I checked the code and the wrappers seems very deep and I didn't see where I can specify the num_workers param. Anyone has any idea about it? Any advice is appreciated!

**Comment by Purfview** [comment, marked as answer]
Try this:

```
 torch.set_num_threads(2)
 pipeline.to(torch.device(device))
 ```

**Reply by treya-lin** [reply]
Thanks. It works!
